contracts/aggregator/common/main.sol

```diff
@@ -0,0 +1,200 @@
+
+    /**
+     * @dev owner gaurd.
+     * @notice owner gaurd.
+     */
+    modifier onlyOwner() {
+        require(msg.sender == owner, "not-owner");
+        _;
+    }
+
+    /**
+     * @dev Update owner.
+     * @notice Update owner.
+     * @param newOwner_ address of new owner.
+     */
+    function updateOwner(address newOwner_) external onlyOwner {
+        address oldOwner_ = owner;
+        owner = newOwner_;
+        emit updateOwnerLog(oldOwner_, newOwner_);
+    }
+
+    /**
+     * @dev Function to add new routes.
+     * @notice Function to add new routes and implementations.
+     * @param _routes routes to add.
+     * @param _impls implementations of their respective routes.
+     */
+    function addNewRoutesAndEnable(
+        uint256[] memory _routes,
+        address[] memory _impls
+    ) public onlyOwner {
+        require(_routes.length == _impls.length, "lengths-dont-match");
+        uint256 length = _routes.length;
+        for (uint256 i = 0; i < length; i++) {
+            require(
+                routeToImpl[_routes[i]] == address(0),
+                "route-already-exists"
+            );
+            routeToImpl[_routes[i]] = _impls[i];
+            routeStatus[_routes[i]] = true;
+            routes.push(_routes[i]);
+        }
+    }
+
+    /**
+     * @dev Function to update existing routes.
+     * @notice Function to update existing routes and implementations.
+     * @param _routes routes to update.
+     * @param _impls implementations of their respective routes.
+     */
+    function updateRouteImplementations(
+        uint256[] memory _routes,
+        address[] memory _impls
+    ) public onlyOwner {
+        require(_routes.length == _impls.length, "lengths-dont-match");
+        uint256 length = _routes.length;
+        for (uint256 i = 0; i < length; i++) {
+            require(
+                routeToImpl[_routes[i]] != address(0),
+                "route-doesnt-exist"
+            );
+            routeToImpl[_routes[i]] = _impls[i];
+        }
+    }
+
+    /**
+     * @dev Function to change route status.
+     * @notice Function to enable and disable routes.
+     * @param _routes routes those status we want to change.
+     * @param _statuses new statuses.
+     */
+    function changeRouteStatus(
+        uint256[] memory _routes,
+        bool[] memory _statuses
+    ) public onlyOwner {
+        require(_routes.length == _statuses.length, "lengths-dont-match");
+        uint256 length = _routes.length;
+        for (uint256 i = 0; i < length; i++) {
+            routeStatus[_routes[i]] = _statuses[i];
+        }
+    }
+
+    /**
+     * @dev Function to delete route.
+     * @notice Function to delete route.
+     * @param _route routes to delete.
+     */
+    function deleteRoute(uint256 _route) public onlyOwner {
+        uint256 length = routes.length;
+        for (uint256 i = 0; i < length; i++) {
+            if (routes[i] == _route) {
+                delete routes[i];
```

Comment:
@shriyatyagii This wont delete it, this will just put `0` at the index. Need to swap it with the last element and pop it.

Reply:
https://blog.finxter.com/how-to-delete-an-element-from-an-array-in-solidity/#:~:text=We%20can%20create%20a%20function,using%20the%20pop()%20method.

Reply:
Oh! I see. Only the value will be deleted. Fixed it!